## Import Library

In [1]:
import numpy as np
import pandas as pd

#---------- For Descriptive Statistics #----------
import math
import statistics
import scipy.stats
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

### Ambil Data

In [2]:
df = pd.read_csv("datasets/nyc-rolling-sales.csv")

# Cek Data
df.head(10)

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0           4        1  ALPHABET CITY   
1           5        1  ALPHABET CITY   
2           6        1  ALPHABET CITY   
3           7        1  ALPHABET CITY   
4           8        1  ALPHABET CITY   
5           9        1  ALPHABET CITY   
6          10        1  ALPHABET CITY   
7          11        1  ALPHABET CITY   
8          12        1  ALPHABET CITY   
9          13        1  ALPHABET CITY   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  BLOCK  \
0  07 RENTALS - WALKUP APARTMENTS                                2A    392   
1  07 RENTALS - WALKUP APARTMENTS                                 2    399   
2  07 RENTALS - WALKUP APARTMENTS                                 2    399   
3  07 RENTALS - WALKUP APARTMENTS                                2B    402   
4  07 RENTALS - WALKUP APARTMENTS                                2A    404   
5  07 RENTALS - WALKUP APARTMENTS                                 2    405   
6  07 RENTALS - WALKUP APARTMENTS                                2B    406   
7  07 RENTALS - WALKUP APARTMENTS                                 2    407   
8  08 RENTALS - ELEVATOR APARTMENTS                               2    379   
9  08 RENTALS - ELEVATOR APARTMENTS                               2    387   

   LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  ...  \
0    6                                  C2            153 AVENUE B  ...   
1   26                                  C7   234 EAST 4TH   STREET  ...   
2   39                                  C7   197 EAST 3RD   STREET  ...   
3   21                                  C4     154 EAST 7TH STREET  ...   
4   55                                  C2  301 EAST 10TH   STREET  ...   
5   16                                  C4  516 EAST 12TH   STREET  ...   
6   32                                  C4            210 AVENUE B  ...   
7   18                                  C7  520 EAST 14TH   STREET  ...   
8   34                                  D5            141 AVENUE D  ...   
9  153                                  D9     629 EAST 5TH STREET  ...   

  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0                 5                 0            5              1633   
1                28                 3           31              4616   
2                16                 1           17              2212   
3                10                 0           10              2272   
4                 6                 0            6              2369   
5                20                 0           20              2581   
6                 8                 0            8              1750   
7                44                 2           46              5163   
8                15                 0           15              1534   
9                24                 0           24              4489   

   GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0               6440       1900                         2   
1              18690       1900                         2   
2               7803       1900                         2   
3               6794       1913                         2   
4               4615       1900                         2   
5               9730       1900                         2   
6               4226       1920                         2   
7              21007       1900                         2   
8               9198       1920                         2   
9              18523       1920                         2   

   BUILDING CLASS AT TIME OF SALE  SALE PRICE            SALE DATE  
0                              C2     6625000  2017-07-19 00:00:00  
1                              C7         -    2016-12-14 00:00:00  
2                              C7         -    2016-12-09 00:00:00  
3                              C4     3936272  2016-09-23 00:00:00  
4                              C2     8000000  2016-11-1

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Unnamed: 0                      84548 non-null  int64 
 1   BOROUGH                         84548 non-null  int64 
 2   NEIGHBORHOOD                    84548 non-null  object
 3   BUILDING CLASS CATEGORY         84548 non-null  object
 4   TAX CLASS AT PRESENT            84548 non-null  object
 5   BLOCK                           84548 non-null  int64 
 6   LOT                             84548 non-null  int64 
 7   EASE-MENT                       84548 non-null  object
 8   BUILDING CLASS AT PRESENT       84548 non-null  object
 9   ADDRESS                         84548 non-null  object
 10  APARTMENT NUMBER                84548 non-null  object
 11  ZIP CODE                        84548 non-null  int64 
 12  RESIDENTIAL UNITS               84548 non-null

In [4]:
df.shape

(84548, 22)

<b>Keterangan</b>

Selanjutnya adalah menghapus kolom EASE-MENT dan Unnamed: 0, dikarenakan data tersebut tidak diperlukan dan tidak ada nilai di dalam kolom tersebut, maka agar mendapatkan hasil yang akurat, maka 2 kolom tersebut dihapus dari tabel asli

In [5]:
# Menghapus Kolom EASE-MENT dan Unnamed: 0, dikarenakan data tidak terlalu diperlukan
df_no_east = df.drop(["Unnamed: 0", "EASE-MENT"], axis=1)

# Cek Data
df_no_east.head(5)

BOROUGH   NEIGHBORHOOD                      BUILDING CLASS CATEGORY  \
0        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
1        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
2        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
3        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
4        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                

  TAX CLASS AT PRESENT  BLOCK  LOT BUILDING CLASS AT PRESENT  \
0                   2A    392    6                        C2   
1                    2    399   26                        C7   
2                    2    399   39                        C7   
3                   2B    402   21                        C4   
4                   2A    404   55                        C2   

                  ADDRESS APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  \
0            153 AVENUE B                      10009                  5   
1   234 EAST 4TH   STREET                      10009                 28   
2   197 EAST 3RD   STREET                      10009                 16   
3     154 EAST 7TH STREET                      10009                 10   
4  301 EAST 10TH   STREET                      10009                  6   

   COMMERCIAL UNITS  TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  \
0                 0            5             1633              6440   
1                 3           31             4616             18690   
2                 1           17             2212              7803   
3                 0           10             2272              6794   
4                 0            6             2369              4615   

   YEAR BUILT  TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE  \
0        1900                          2                             C2   
1        1900                          2                             C7   
2        1900                          2                             C7   
3        1913                          2                             C4   
4        1900                          2                             C2   

  SALE PRICE            SALE DATE  
0    6625000  2017-07-19 00:00:00  
1        -    2016-12-14 00:00:00  
2        -    2016-12-09 00:00:00  
3    3936272  2016-09-23 00:00:00  
4    8000000  2016-11-17 00:00:00

In [6]:
df_no_east.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   BOROUGH                         84548 non-null  int64 
 1   NEIGHBORHOOD                    84548 non-null  object
 2   BUILDING CLASS CATEGORY         84548 non-null  object
 3   TAX CLASS AT PRESENT            84548 non-null  object
 4   BLOCK                           84548 non-null  int64 
 5   LOT                             84548 non-null  int64 
 6   BUILDING CLASS AT PRESENT       84548 non-null  object
 7   ADDRESS                         84548 non-null  object
 8   APARTMENT NUMBER                84548 non-null  object
 9   ZIP CODE                        84548 non-null  int64 
 10  RESIDENTIAL UNITS               84548 non-null  int64 
 11  COMMERCIAL UNITS                84548 non-null  int64 
 12  TOTAL UNITS                     84548 non-null

<b>Keterangan</b>

Mengecek tipe data setiap kolom, agar lebih mempermudah untuk mengelompokkan setiap kolom dengan tipe data yang sama dan dengan tujuan untuk mempermudah dalam pengerjaan

In [7]:
# Mengelompokan data dengan type data object
data_cat_obj = [
    "NEIGHBORHOOD", 
    "BUILDING CLASS CATEGORY", 
    "TAX CLASS AT PRESENT",
    "BLOCK",
    "LOT",
    "BUILDING CLASS AT PRESENT",
    "ADDRESS",
    "APARTMENT NUMBER",
    "BUILDING CLASS AT TIME OF SALE",
    "SALE DATE"
]
for data_group in data_cat_obj:
    df_no_east[data_group] = df_no_east[data_group].astype('category')

# Mengelompokan data dengan type data integer dan mengubahnya ke numeric
data_cat_int = [
    "BOROUGH",
    "SALE PRICE",
    "LAND SQUARE FEET",
    "GROSS SQUARE FEET",
    
]
for data_group in data_cat_int:
    df_no_east[data_group] = pd.to_numeric(df_no_east[data_group], errors='coerce')

# Mengubah type data SALE DATE dari Object menjadi Date
df_no_east["SALE DATE"] = pd.to_datetime(df_no_east["SALE DATE"])

<b>Catatan</b>

Untuk menghindari Error yang terjadi pada kolom SALE PRICE yang memiliki nilai (-), maka bisa mengabaikannya dengan menambahkan errors="coerce", yang dimana nantinya isi yang memiliki (-) akan diubah menjadi NaN

<b>More Information</b>

Parameters
argscalar, list, tuple, 1-d array, or Series
Argument to be converted.

errors{‘ignore’, ‘raise’, ‘coerce’}, default ‘raise’
* If ‘raise’, then invalid parsing will raise an exception.
* If ‘coerce’, then invalid parsing will be set as NaN.
* If ‘ignore’, then invalid parsing will return the input.

<b>Keterangan</b>

Selanjutnya adalah mengecek jumlah data yang memliki nilai null

In [8]:
# Mengecek nilai null
df_no_east.isnull().sum()

BOROUGH                               0
NEIGHBORHOOD                          0
BUILDING CLASS CATEGORY               0
TAX CLASS AT PRESENT                  0
BLOCK                                 0
LOT                                   0
BUILDING CLASS AT PRESENT             0
ADDRESS                               0
APARTMENT NUMBER                      0
ZIP CODE                              0
RESIDENTIAL UNITS                     0
COMMERCIAL UNITS                      0
TOTAL UNITS                           0
LAND SQUARE FEET                  26252
GROSS SQUARE FEET                 27612
YEAR BUILT                            0
TAX CLASS AT TIME OF SALE             0
BUILDING CLASS AT TIME OF SALE        0
SALE PRICE                        14561
SALE DATE                             0
dtype: int64

<b>Keterangan</b>

Dari hasil pengecekan tersebut terdapat nilai null pada koilom "LAND SQUARE FEET sebanyak = 26252, GROSS SQUARE FEET sebanyak : 27612, dan SALE PRICE sebanyak : 14561"

Dari jumlah nilai null sebanyak itu maka kita dapat mengganti nilai null tersebut dengan menggunakan MEAN dari setiap kolom tersebut agar data yang dihasilkan lebih akurat

In [9]:
df_no_east['LAND SQUARE FEET'].fillna(df_no_east['LAND SQUARE FEET'].mean(), inplace=True)
df_no_east['GROSS SQUARE FEET'].fillna(df_no_east['GROSS SQUARE FEET'].mean(), inplace=True)
df_no_east['SALE PRICE'].fillna(df_no_east['SALE PRICE'].mean(), inplace=True)

In [10]:
df_no_east[['LAND SQUARE FEET', 'GROSS SQUARE FEET', 'SALE PRICE']].isnull().sum()

LAND SQUARE FEET     0
GROSS SQUARE FEET    0
SALE PRICE           0
dtype: int64

<b>Keterangan</b>

Dari pengisian nilai null tersebut dengan menggunakan mean dari setiap kolom maka di dapatkan hasil berupa jumlah null dari kolom tersebut berjumlah 0

In [11]:
df_no_east.head()

BOROUGH   NEIGHBORHOOD                      BUILDING CLASS CATEGORY  \
0        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
1        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
2        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
3        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                
4        1  ALPHABET CITY  07 RENTALS - WALKUP APARTMENTS                

  TAX CLASS AT PRESENT BLOCK LOT BUILDING CLASS AT PRESENT  \
0                   2A   392   6                        C2   
1                    2   399  26                        C7   
2                    2   399  39                        C7   
3                   2B   402  21                        C4   
4                   2A   404  55                        C2   

                  ADDRESS APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  \
0            153 AVENUE B                      10009                  5   
1   234 EAST 4TH   STREET                      10009                 28   
2   197 EAST 3RD   STREET                      10009                 16   
3     154 EAST 7TH STREET                      10009                 10   
4  301 EAST 10TH   STREET                      10009                  6   

   COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  GROSS SQUARE FEET  \
0                 0            5            1633.0             6440.0   
1                 3           31            4616.0            18690.0   
2                 1           17            2212.0             7803.0   
3                 0           10            2272.0             6794.0   
4                 0            6            2369.0             4615.0   

   YEAR BUILT  TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE  \
0        1900                          2                             C2   
1        1900                          2                             C7   
2        1900                          2                             C7   
3        1913                          2                             C4   
4        1900                          2                             C2   

     SALE PRICE  SALE DATE  
0  6.625000e+06 2017-07-19  
1  1.276456e+06 2016-12-14  
2  1.276456e+06 2016-12-09  
3  3.936272e+06 2016-09-23  
4  8.000000e+06 2016-11-17

In [12]:
df_no_east['RESIDENTIAL UNITS'].value_counts()

1      34722
0      24783
2      16049
3       4608
4       1346
       ...  
179        1
75         1
193        1
153        1
128        1
Name: RESIDENTIAL UNITS, Length: 176, dtype: int64

<b>Keterangan</b>

Selanjutnya adalah melakukan ekplorasi data yang ada dengan menggunakan .describe()

In [13]:
df_no_east[df_no_east['COMMERCIAL UNITS']==0].describe()

BOROUGH      ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
count  79429.000000  79429.000000       79429.000000           79429.0   
mean       3.004797  10721.690503           1.717257               0.0   
std        1.299437   1320.121756          10.146362               0.0   
min        1.000000      0.000000           0.000000               0.0   
25%        2.000000  10304.000000           0.000000               0.0   
50%        3.000000  11209.000000           1.000000               0.0   
75%        4.000000  11357.000000           2.000000               0.0   
max        5.000000  11694.000000         889.000000               0.0   

        TOTAL UNITS  LAND SQUARE FEET  GROSS SQUARE FEET    YEAR BUILT  \
count  79429.000000      7.942900e+04       7.942900e+04  79429.000000   
mean       1.750217      3.423023e+03       2.964567e+03   1780.395787   
std       10.142353      2.394616e+04       9.865621e+03    551.999810   
min        0.000000      0.000000e+00       0.000000e+00      0.000000   
25%        1.000000      2.000000e+03       1.312000e+03   1920.000000   
50%        1.000000      3.263000e+03       2.400000e+03   1940.000000   
75%        2.000000      3.941676e+03       4.045707e+03   1967.000000   
max      889.000000      4.252327e+06       1.378125e+06   2017.000000   

       TAX CLASS AT TIME OF SALE    SALE PRICE  
count               79429.000000  7.942900e+04  
mean                    1.584258  1.040170e+06  
std                     0.720488  3.028813e+06  
min                     1.000000  0.000000e+00  
25%                     1.000000  3.000000e+05  
50%                     1.000000  6.550000e+05  
75%                     2.000000  1.276456e+06  
max                     4.000000  3.450000e+08

In [14]:
df_no_east[df_no_east['RESIDENTIAL UNITS']==0].describe()

BOROUGH      ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
count  24783.000000  24783.000000            24783.0      24783.000000   
mean       2.537869  10318.242182                0.0          0.492515   
std        1.335198   2136.962746                0.0         16.073953   
min        1.000000      0.000000                0.0          0.000000   
25%        1.000000  10023.000000                0.0          0.000000   
50%        3.000000  11004.000000                0.0          0.000000   
75%        4.000000  11354.000000                0.0          0.000000   
max        5.000000  11694.000000                0.0       2261.000000   

        TOTAL UNITS  LAND SQUARE FEET  GROSS SQUARE FEET    YEAR BUILT  \
count  24783.000000      2.478300e+04       2.478300e+04  24783.000000   
mean       0.598152      5.465813e+03       5.431326e+03   1674.012145   
std       16.073655      5.098233e+04       3.262132e+04    680.924383   
min        0.000000      0.000000e+00       0.000000e+00      0.000000   
25%        0.000000      2.047000e+03       2.775000e+02   1921.000000   
50%        0.000000      3.941676e+03       4.045707e+03   1950.000000   
75%        0.000000      3.941676e+03       4.045707e+03   1962.000000   
max     2261.000000      4.252327e+06       1.617206e+06   2017.000000   

       TAX CLASS AT TIME OF SALE    SALE PRICE  
count               24783.000000  2.478300e+04  
mean                    2.423879  1.575554e+06  
std                     0.913323  1.807579e+07  
min                     1.000000  0.000000e+00  
25%                     2.000000  2.150000e+05  
50%                     2.000000  5.200000e+05  
75%                     2.000000  1.276456e+06  
max                     4.000000  2.210000e+09

In [15]:
df_no_east.value_counts(["BOROUGH"])

BOROUGH
4          26736
3          24047
1          18306
5           8410
2           7049
dtype: int64

In [16]:
df_no_east.groupby('BOROUGH').mean()

ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
BOROUGH                                                                   
1         9913.852125           2.307495          0.299683     2.649623   
2        10361.799404           3.436658          0.160874     3.603774   
3        11073.456938           2.035639          0.147960     2.220651   
4        11300.420669           1.701489          0.206351     1.938996   
5        10039.674197           1.227586          0.079667     1.309869   

         LAND SQUARE FEET  GROSS SQUARE FEET   YEAR BUILT  \
BOROUGH                                                     
1             4114.800092        7150.756683  1706.773845   
2             3947.091502        4524.350262  1745.715279   
3             2166.259825        2570.354597  1767.358673   
4             4167.674952        3568.246454  1852.045631   
5             7918.337390        2622.190827  1868.961237   

         TAX CLASS AT TIME OF SALE    SALE PRICE  
BOROUGH                                           
1                         2.130176  2.902477e+06  
2                         1.525606  5.901936e+05  
3                         1.661455  8.344884e+05  
4                         1.513278  9.063757e+05  
5                         1.186207  7.525600e+05

In [17]:
df_no_east.groupby('BOROUGH').apply(pd.DataFrame.mode).reset_index(drop=True)

BOROUGH             NEIGHBORHOOD  \
0        1  UPPER EAST SIDE (59-79)   
1        2                RIVERDALE   
2        3       BEDFORD STUYVESANT   
3        4           FLUSHING-NORTH   
4        5              GREAT KILLS   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT BLOCK LOT  \
0  13 CONDOS - ELEVATOR APARTMENTS                                2    16   1   
1  02 TWO FAMILY DWELLINGS                                        1  3944   1   
2  02 TWO FAMILY DWELLINGS                                        1  2135   1   
3  01 ONE FAMILY DWELLINGS                                        1  5066   1   
4  01 ONE FAMILY DWELLINGS                                        1  2450   1   

  BUILDING CLASS AT PRESENT                 ADDRESS APARTMENT NUMBER  \
0                        R4  169 WEST 95TH   STREET                    
1                        D4     2287 JOHNSON AVENUE                    
2                        R4         429 KENT AVENUE                    
3                        A1      131-05 40TH   ROAD                    
4                        A5   1235 FOREST HILL ROAD                    

   ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
0     10023                  0                 0            1   
1     10462                  1                 0            1   
2     11201                  1                 0            1   
3     11354                  1                 0            1   
4     10314                  1                 0            1   

   LAND SQUARE FEET  GROSS SQUARE FEET  YEAR BUILT  TAX CLASS AT TIME OF SALE  \
0       3941.675775        4045.707479           0                          2   
1          0.000000           0.000000           0                          1   
2          0.000000           0.000000           0                          1   
3       3941.675775        4045.707479        1920                          1   
4       3941.675775        4045.707479        1970                          1   

  BUILDING CLASS AT TIME OF SALE    SALE PRICE  SALE DATE  
0                             R4  1.276456e+06 2017-08-07  
1                             D4  0.000000e+00 2017-01-18  
2                             R4  0.000000e+00 2016-10-06  
3                             A1  1.276456e+06 2016-11-10  
4                             A5  1.276456e+06 2017-06-15

In [18]:
df_no_east.groupby('BOROUGH').var()

ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
BOROUGH                                                                   
1        1.152993e+06         494.548162         42.708328   543.256962   
2        1.061562e+06         560.408622          0.769235   576.031263   
3        1.627799e+06          73.069968          2.696896    75.907229   
4        8.154062e+05         329.581619        206.971706   542.436937   
5        2.695262e+06           5.259888          3.919446     9.045604   

         LAND SQUARE FEET  GROSS SQUARE FEET     YEAR BUILT  \
BOROUGH                                                       
1            2.673569e+07       1.400474e+09  420445.857838   
2            3.420048e+08       8.726635e+08  348076.960778   
3            1.980388e+08       2.348546e+08  313207.618126   
4            1.389233e+09       1.155549e+09  181040.864946   
5            6.866474e+09       1.539069e+08  188223.832009   

         TAX CLASS AT TIME OF SALE    SALE PRICE  
BOROUGH                                           
1                         0.301491  4.587355e+14  
2                         0.700288  7.745411e+12  
3                         0.788360  1.549063e+13  
4                         0.645719  7.274834e+12  
5                         0.409847  2.694457e+12

In [19]:
df_no_east.groupby('BOROUGH').std()

ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
BOROUGH                                                                  
1        1073.775249          22.238439          6.535161    23.307873   
2        1030.321292          23.672951          0.877060    24.000651   
3        1275.852225           8.548097          1.642223     8.712475   
4         902.998467          18.154383         14.386511    23.290276   
5        1641.725274           2.293445          1.979759     3.007591   

         LAND SQUARE FEET  GROSS SQUARE FEET  YEAR BUILT  \
BOROUGH                                                    
1             5170.656325       37422.911692  648.417965   
2            18493.371717       29540.878057  589.980475   
3            14072.627073       15324.967190  559.649549   
4            37272.411205       33993.367954  425.488972   
5            82864.192024       12405.921583  433.847706   

         TAX CLASS AT TIME OF SALE    SALE PRICE  
BOROUGH                                           
1                         0.549082  2.141811e+07  
2                         0.836832  2.783058e+06  
3                         0.887896  3.935813e+06  
4                         0.803566  2.697190e+06  
5                         0.640193  1.641480e+06

In [20]:
# Melakukan Grade data
df_no_east.groupby('BOROUGH').agg({'SALE PRICE': [np.mean, np.std, np.size]})


SALE PRICE                     
                 mean           std   size
BOROUGH                                   
1        2.902477e+06  2.141811e+07  18306
2        5.901936e+05  2.783058e+06   7049
3        8.344884e+05  3.935813e+06  24047
4        9.063757e+05  2.697190e+06  26736
5        7.525600e+05  1.641480e+06   8410

In [21]:
mean_borough = 2.902477e+06
sd_borough = 2.141811e+07
n_borough = 18306
z = 1.96 # z-score untuk confidence level 95%

<b>Menghitung dengan memasukan rumus standard error</b>

Rumus :

se = sd / np.sqrt(jumlah keseluruhan data)

<b>Keterangan:</b>
* se = standard error
* sd = standard devisiasi

In [22]:
se = sd_borough / np.sqrt(n_borough)

# Cetak Hasil
print("Hasil standard error adalah:", se)

Hasil standard error adalah: 158301.276054372


<b>Melakukan Contsruct CI</b>

Rumus :

lcb = mean_objek - z * se --> untuk lower limit of the CI

ucb = mean_objek + z * se --> untuk upper limit of the CI

In [23]:
lcb = mean_borough - z * se
ucb = mean_borough + z * se

print("Hasil Lower limit adalah:", lcb)
print("-------------------------")
print("Hasil Upper limit adalah:", ucb)

Hasil Lower limit adalah: 2592206.4989334308
-------------------------
Hasil Upper limit adalah: 3212747.5010665692


In [24]:
# Grading Data
grade_borough_1 = df_no_east[df_no_east["BOROUGH"] == 1]
grade_borough_2 = df_no_east[df_no_east["BOROUGH"] == 3]

# Menghitung panjang nilai
# Data 1
num_grade_borough_1 = len(grade_borough_1)
# Data 2
num_grade_borough_2 = len(grade_borough_2)

# Menghitung Mean
# Data 1
mean_grade_borough_1 = grade_borough_1["SALE PRICE"].mean()
# Data 2
mean_grade_borough_2 = grade_borough_2["SALE PRICE"].mean()

# Menghitung Standard Deviasi
# Data 1
std_grade_borough_1 = grade_borough_1["SALE PRICE"].std()
# Data 2
std_grade_borough_2 = grade_borough_2["SALE PRICE"].std()


In [25]:
print("Hasil perhitungan Data 1:",num_grade_borough_1, mean_grade_borough_1, std_grade_borough_1)
print("")
print("Hasil perhitungan Data 2:",num_grade_borough_2, mean_grade_borough_2, std_grade_borough_2)

Hasil perhitungan Data 1: 18306 2902476.5473536155 21418111.93479176

Hasil perhitungan Data 2: 24047 834488.3717719465 3935813.428569773


In [26]:
# Summary
sm.stats.ztest(grade_borough_1["SALE PRICE"].dropna(), grade_borough_2["SALE PRICE"].dropna(), alternative="two-sided")

(14.651229205435005, 1.323003784566643e-48)